### ISSUE TITLE
Font on login screen not shown with Microsoft RDP Beta for Mac

### ISSUE THREAD

**Issue by proski** [neutrinolabs/xrdp#643]
The issue exists with the current xrdp from git and the current MS RDP Beta (8.2.30). All older versions of MS RDP Beta I tried (at least from 8.2.10) had that issue as well. I'm not aware of any other client having that issue. While it may be a bug in MS RDP Beta, let's see if anything can be done in xrdp to work it around.
![login](https://cloud.githubusercontent.com/assets/1317472/22274905/f225bbae-e25d-11e6-94ea-f6b3090647a2.png)

**Comment by metalefty** [comment]
iOS client has the same issue #367.

**Comment by proski** [comment]
OK, then fixing it becomes more interesting. It turns out the client writes some interesting information to the console.
```
2017-01-24 18:39:00.979 Microsoft Remote Desktop Beta[19588:695242] Reachability Flag Status: R ------- networkStatusForFlags
2017-01-24 18:39:01.529 Microsoft Remote Desktop Beta[19588:695216] [HockeySDK] -[BITCrashManager invokeDelayedProcessing]/875 [HockeySDK] WARNING: Another exception handler was added. If this invokes any kind exit() after processing the exception, which causes any subsequent error handler not to be invoked, these crashes will NOT be reported to HockeyApp!
2017-01-24 18:39:10.053 Microsoft Remote Desktop Beta[19588:695216] [PLATFORMSTUB] RDPAPI_CreateInstance unimplemented! in platform_stubs.cpp, line 178
2017-01-24 18:39:10.053 Microsoft Remote Desktop Beta[19588:695216] [RDP_WAN] Failed to create RdpEventLogSession in ConnectionMonitor.cpp, line 474
2017-01-24 18:39:10.053 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] ReadMachinePolicyDWORD failed! in autil.cpp, line 2850
2017-01-24 18:39:10.053 Microsoft Remote Desktop Beta[19588:695216] [PLATFORMSTUB] RDPAPI_CreateInstance unimplemented! in platform_stubs.cpp, line 178
2017-01-24 18:39:10.053 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed to create event log session in coreapi.cpp, line 3101
2017-01-24 18:39:10.230 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Event Source: 0x27 not found
2017-01-24 18:39:10.230 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Unable to find event source. bailing
2017-01-24 18:39:10.230 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] SinkMap Bind failed in basecoreapieventsink.cpp, line 105
2017-01-24 18:39:10.230 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed to initialize sink in basecoreapieventsink.cpp, line 31
2017-01-24 18:39:10.230 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed to create CoreEvent sink in basecoreapi.cpp, line 2530
2017-01-24 18:39:10.230 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Null Smartcard redirection adaptor. in RdpXUClientDeviceRDManager.cpp, line 98
2017-01-24 18:39:10.231 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] GetModuleFileName() error, could not retrieve path
2017-01-24 18:39:10.231 Microsoft Remote Desktop Beta[19588:695216] [PLATFORMSTUB] RDPAPI_CreateInstance unimplemented! in platform_stubs.cpp, line 178
2017-01-24 18:39:10.232 Microsoft Remote Desktop Beta[19588:695216] [PLATFORMSTUB] PAL_System_GetWindowsProductId unimplemented! in platform_stubs.cpp, line 129
2017-01-24 18:39:10.232 Microsoft Remote Desktop Beta[19588:695216] [RDP_WAN] No server addresses were set in the transport props. in TsTcpTransport.cpp, line 163
2017-01-24 18:39:10.232 Microsoft Remote Desktop Beta[19588:695216] [RDP_WAN] Did not add server addresses to property store. 0x80070057 in TsTcpTransport.cpp, line 288
2017-01-24 18:39:10.362 Microsoft Remote Desktop Beta[19588:695216] [PLATFORMSTUB] RDPAPI_CreateInstance unimplemented! in platform_stubs.cpp, line 178
2017-01-24 18:39:10.362 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Unable to pop filter 0x70000e7fb990 from head
2017-01-24 18:39:10.362 Microsoft Remote Desktop Beta[19588:695216] [PLATFORMSTUB] PAL_System_GetWindowsProductId unimplemented! in platform_stubs.cpp, line 129
2017-01-24 18:39:10.362 Microsoft Remote Desktop Beta[19588:695216] [RDP_WAN] No server addresses were set in the transport props. in TsTcpTransport.cpp, line 163
2017-01-24 18:39:10.362 Microsoft Remote Desktop Beta[19588:695216] [RDP_WAN] Did not add server addresses to property store. 0x80070057 in TsTcpTransport.cpp, line 288
2017-01-24 18:39:10.484 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] No data of type 0xc04
2017-01-24 18:39:10.484 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] No data of type 0xc08
2017-01-24 18:39:10.848 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Ignoring unknown capset received 6.
2017-01-24 18:39:10.848 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Recvd Cap Size from Server: 407
2017-01-24 18:39:10.848 Microsoft Remote Desktop Beta[19588:695216] [PLATFORMSTUB] RDPAPI_CreateInstance unimplemented! in platform_stubs.cpp, line 178
2017-01-24 18:39:10.848 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Class allocation failed in coreapi.cpp, line 2862
2017-01-24 18:39:10.848 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] GetImageDecompressor failed! in ccfsm.cpp, line 2011
2017-01-24 18:39:10.848 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] PopulateCacCodecCaps failed!
2017-01-24 18:39:10.848 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Bitmap cache capabilities not found
2017-01-24 18:39:13.498 Microsoft Remote Desktop Beta[19588:695216] [TAP] Failed to create the perf logger manager. in CRDPPerfLoggerAdaptor.cpp, line 47
2017-01-24 18:39:13.498 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed to create logger instance in PerformanceCounter.cpp, line 77
2017-01-24 18:39:13.499 Microsoft Remote Desktop Beta[19588:695216] [TAP] Failed to create the perf logger manager. in CRDPPerfLoggerAdaptor.cpp, line 47
2017-01-24 18:39:13.499 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed to create logger instance in PerformanceCounter.cpp, line 77
2017-01-24 18:39:13.499 Microsoft Remote Desktop Beta[19588:695216] [TAP] Failed to create the perf logger manager. in CRDPPerfLoggerAdaptor.cpp, line 47
2017-01-24 18:39:13.499 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed to create logger instance in PerformanceCounter.cpp, line 77
2017-01-24 18:39:13.499 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.500 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.500 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.500 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.500 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.501 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.501 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.501 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.501 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.502 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.502 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.502 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.502 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.502 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.502 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.503 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.503 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.503 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.503 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.503 Microsoft Remote Desktop Beta[19588:695216] [TAP] Failed to create the perf logger manager. in CRDPPerfLoggerAdaptor.cpp, line 47
2017-01-24 18:39:13.503 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed to create logger instance in PerformanceCounter.cpp, line 77
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.528 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] IsValidGlyphCacheIDIndex failed in TsGfxGlyph.cpp, line 548
2017-01-24 18:39:13.529 Microsoft Remote Desktop Beta[19588:695216] [-legacy-] Failed ProcessCacheGlyphOrder in TsGfxSurface.cpp, line 2497
2017-01-24 18:39:21.448 Microsoft Remote Desktop Beta[19588:695216] [RDP_SYSTEM] Tearing down thread 1 in RdpMacSystemPALThread.cpp, line 580
2017-01-24 18:39:47.455 Microsoft Remote Desktop Beta[19588:695216] [RDP_SYSTEM] Tearing down thread 3 in RdpMacSystemPALThread.cpp, line 580
```

**Comment by derekschrock** [comment]
This continues to be an issue on MS RDP macOS client Version 10.1.1 (833)

**Comment by leedor** [comment]
... and continues to be an issue on MS RDP macOS client Version 10.1.3 (843)

**Comment by derekschrock** [comment]
I'm pretty sure this issue can be closed.  MS has acknowledged this issue with their client.   Are we sure this is an xrdp issue?

**Comment by jsorg71** [comment]
This was an xrdp issue but it was fixed when we add libpainter.  If you still see it, report the xrdp version and client.

**Comment by derekschrock** [comment]
MS Remote Desktop macOS client Version 10.1.4 (850)
xrdp commit e16029c

**Comment by merced317** [comment]
I'm still experiencing this issue along with a black box around the mouse after login.
MS Remote Desktop macOS client Version 10.1.5 (866)
xrdp version 0.9.4 from https://launchpad.net/~hermlnx/+archive/ubuntu/xrdp

**Comment by moobyfr** [comment]
not sure it the issue is the same, but I see black square with current master from freerdp

**Comment by derekschrock** [comment]
No fonts with MS RDP Beta client Version 10.1.6 (910) at the sesman login

**Comment by akohlsmith** [comment]
Also seeing this with Jump Desktop on OSX.

**Comment by shamilovtim** [comment]
Currently unable to use RDP due to this

**Comment by Webern** [comment]
No text on login screen
```
xrdp: A Remote Desktop Protocol server.
Copyright (C) Jay Sorg 2004-2014
See http://www.xrdp.org for more information.
Version 0.9.5
```
```
Microsoft Remote Desktop 10.8.1 (983)
macOS 10.13.4
```

**Comment by metalefty** [comment]
Enough reports collected.

**Comment by metalefty** [comment]
Now the old version is deprecated. The beta version is released as version 10.
We need to work on this because ver 10 is the only recommended version to Mac.

@speidy @jsorg71

**Comment by metalefty** [comment]
See also: https://cloudblogs.microsoft.com/enterprisemobility/2018/07/13/deprecation-of-the-remote-desktop-8-0-client-for-macos/

**Comment by metalefty** [comment]
Hi everyone, the issue is fixed by #1235. Will be included in next release v0.9.9.
